### PR TITLE
Abstract interface of the DA service.

### PR DESCRIPTION
**Pull request by steelgeek091**
An abstract interface of the DA service has been added: The DA server is an independent service that handles user storage requests, stores user data to one or more backend, and returns response data from the DA committee.

The following issue link provides a rough explanation of the entire working process:
https://github.com/rooch-network/rooch/issues/134

### CODE REVIEW

File: moveos/moveos-client/src/client.rs [hunk 1]
```diff
@@ -197,3 +197,46 @@ struct RPCResponse<T> {
...
+struct DAGetResponse {
+    data: Vec<u8>,
+}
+
+struct DAStatusResponse {
+    status_code: u64,
+    status_data: Vec<u8>,
+}
+
+trait DAServer {
```
Comment (jolestar) [inline moveos/moveos-client/src/client.rs:238]: I think the DAServer should be in a crate like `crate/rooch-da`, not in the `moveos-client`

Comment (steelgeek091) [reply moveos/moveos-client/src/client.rs:238]: Okay, I will resubmit it again.

**Comment by templexxx** [comment]
You could add some description of this design, especially the usage.

1. components: client, server, backend
2. how to use server and backend hide the DA network's details
3. how celesita/ipfs/...  works as backend

or u can explain it in issue, then link here

**Comment by jolestar** [comment]
I think we currently only need to define the DA client interface, and the implementation is a wrap of the DA's client(like Celestia or Arweave or ETH)?

**Comment by jolestar** [comment]
Is this PR included in #136?

**Comment by steelgeek091** [comment]
> I think we currently only need to define the DA client interface, and the implementation is a wrap of the DA's client(like Celestia or Arweave or ETH)?



> Is this PR included in #136?

I just found out that #136 includes this pull request. Please refer to #136 for subsequent changes. Thank you.

**Comment by jolestar** [comment]
I close this.

**Pull request closed without merging**